libs/island-ui/core/src/lib/Text/Text.tsx

```diff
@@ -45,58 +45,67 @@ export interface TextProps {
   lineHeight?: keyof typeof lineHeightStyles
 }
 
-export const Text = ({
-  id,
-  children,
-  color,
-  truncate,
-  paddingTop,
-  paddingBottom,
-  paddingY,
-  marginTop,
-  marginBottom,
-  marginY,
-  fontWeight,
-  lineHeight,
-  variant = 'default',
-  as = 'p',
-}: TextProps) => {
-  const { linkRenderer } = useContext(LinkContext)
-  return (
-    <Box
-      id={id}
-      component={as}
-      marginTop={marginTop}
-      marginBottom={marginBottom}
-      marginY={marginY}
-      paddingTop={paddingTop}
-      paddingBottom={paddingBottom}
-      paddingY={paddingY}
-      className={cn(base, {
-        [variantStyles[variant!]]: variant,
-        [colors[color!]]: color,
-        [fontWeightStyles[fontWeight!]]: fontWeight,
-        [lineHeightStyles[lineHeight!]]: lineHeight,
-        [defaultFontWeights[variant!]]: variant && !fontWeight,
-        [defaultLineHeights[variant!]]: variant && !lineHeight,
-        [truncateStyle]: truncate,
-      })}
-    >
-      {React.Children.map<React.ReactNode, React.ReactNode>(
-        children,
-        (child: any) => {
-          if (child?.props?.href && child?.props?.as) {
-            // Checking to see if the child  using "href" and "as" props, which indicates it
-            // is (most likely) a next.js link since the linkRenderer breaks this functionality.
-            // TODO: Make linkRenderer handle next.js links.
-            return child
-          } else if (child?.props?.href && typeof linkRenderer === 'function') {
-            return linkRenderer(child.props.href, child.props.children)
-          }
+export const Text = forwardRef<any, TextProps>(
```

Comment:
Doesn't work if you replace any with `typeof Box`?

Reply:
`typeof Box` is not working, I probably should be using `HTMLElement` I'll change it